clippy_lints/src/methods/mod.rs

```diff
@@ -738,6 +738,51 @@ declare_clippy_lint! {
     "using `filter_map` when a more succinct alternative exists"
 }
 
+/// **What it does:** Checks for `into_iter` calls on types which should be replaced by `iter` or
+/// `iter_mut`.
+///
+/// **Why is this bad?** Arrays and `PathBuf` do not yet have an `into_iter` method which move out
+/// their content into an iterator. Calling `into_iter` instead just forwards to `iter` or
+/// `iter_mut` due to auto-referencing, of which only yield references. Furthermore, when the
```

Comment:
"of which only yield"

Reply:
@oli-obk sorry I'm not sure what do you mean

Reply:
This sentence doesn't parse for me. I'm not sure what you're trying to say.

Reply:
IIRC `iter_mut` doesn't directly apply, because it will always try and succeed with the immutable reference first, matching only `iter`.

Perhaps something like: "Auto-referencing resolves the `into_iter` call onto a container reference instead, like `<&[T; N] as IntoIterator>::into_iter`, which just iterates over item references like calling `iter` would."